### ISSUE TITLE
Relative Path Issues

### ISSUE THREAD

**Issue by crutchcorn** [ksoichiro/rehype-img-size#4]
Hey! So I'm working on porting this website to NextJS and intend on using this package to make that happen!

https://github.com/unicorn-utterances/unicorn-utterances/

However, I would like to keep our markdown as-is if possible. Two path types that aren't currently supported are:

```
![](./some_relative_path.img)
```

And

```
![](some_relative_path.img)
```

Both of which are supported from markdown files. I was wondering if it would be okay for me to open a PR to fix these two instance. Here's the code I was thinking of adding:

```javascript
        const shouldJoin =
            src.startsWith('/') ||
            src.startsWith('./') ||
            /^\w/.exec(src);

        if (dir && shouldJoin) {
            src = path.join(dir, src);
        }
```

However, it's worth mentioning that there's questionable reason to check for `\w`, since it might just be best to do a `if (dir)` check instead and remove path checking

**Comment by ksoichiro** [comment]
Sorry for my late reply 🙇 

I think I haven't fully understood what the second one exactly means,

```
![](some_relative_path.img)
```

but supporting `./` or `../` seems good.
We should use `path.resolve()` instead of `path.join()` maybe.

About your suggestion, I think checking with `/\w/` might cause unwanted result when `src` is a complete (external) URL.
Following condition will be evaluated as true, but we don't want to join `dir` in this case.

```
if (/^\w/.exec('https://example.com/img.png')) {
```

In other words, if those exceptions are handled appropriately, I think we can simplify `shouldJoin` (i.e. remove current path checking).

**Comment by crutchcorn** [comment]
> but we don't want to join dir in this case.
> if (/^\w/.exec('https://example.com/img.png')) {

Right right. Since opening this issue, this is what I've done in the fork I'm using for production is the following:

```js
const absolutePathRegex = /^(?:[a-z]+:)?\/\//;

export function getImageSize(src, dir) {
  if (absolutePathRegex.exec(src)) {
    return
  }
  // Treat `/` as a relative path, according to the server
  const shouldJoin = !path.isAbsolute(src) || src.startsWith('/');

  if (dir && shouldJoin) {
    src = path.join(dir, src);
  }
  return sizeOf(src)
}
```

This avoids anything like:

```
http://
https://
ftp://
//
```

But still supports:

```
/server/relative/path/links
filename.txt
./filename.txt
../path/filename.txt
```

This feels like simpler logic with more edgecases handled IMO. What do you think?

**Comment by ksoichiro** [comment]
Great! 
> This feels like simpler logic with more edgecases handled IMO

Yes I completely agree with you 🙂

**Comment by crutchcorn** [comment]
Awesome! :D I'll make a PR right now!

Thanks so much for being so amicable towards these changes!

**Comment by ksoichiro** [comment]
Merged #9 and released as v1.0.0. Thank you!
